fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/dao/impl/FHIRDbDAOImpl.java

```diff
@@ -406,4 +508,28 @@ protected FHIRPersistenceDBConnectException buildFHIRPersistenceDBConnectExcepti
         }
         return strValues;
     }
+
+    /**
+     * Get the translator appropriate for the flavor of database we are using
+     * @return
+     */
+    protected IDatabaseTranslator getTranslator() {
+        IDatabaseTranslator result;
+
+        switch (this.flavor.getType()) {
+        case DERBY:
+            result = new DerbyTranslator();
+            break;
+        case DB2:
+            result = new Db2Translator();
+            break;
+        case POSTGRESQL:
+            result = new PostgresTranslator();
+            break;
+        default:
+            throw new IllegalStateException("DbType not supported: " + this.flavor.getType());
+        }
+
+        return result;
+    }
```

Comment:
we use this in so many spots, maybe it warrants some refactoring into a helper class?

Reply:
```
    protected IDatabaseTranslator getTranslator() {
        return DatabaseTranslatorFactory.getTranslator(this.flavor.getType());
    }
```